### ISSUE TITLE
Improve Documentation

### ISSUE THREAD

**Issue by Batuu13** [Nozbe/WatermelonDB#832]
Hey guys, 

I've been having a hard time implementing stuff from the documentation. Most of the documentation gives a brief look at what is available but there is no detailed documentation. Even a simple relation is missing. We have post-author-comment example with missing schemas, missing models ( I mean fully, not partially because this is not working). I would love to contribute but first I need to learn how this works. All of the examples are using "*_id" for relations but we don't know if this means the original "id" column or it is a self-generated custom "*_id" column. I think documentation is super important for the newcomers if you guys want this project to grow!

I hope my positive criticism efforts won't get unnoticed, thank you!

**Comment by Unforgiven-wanda** [comment]
This one criticism is my one and only reason why I haven't yet used WatermelonDB in any project so far, even though I'd like to. I imagine myself getting stuck on a common-use case 2 days in and sifting through the issues and I just can't, not when there are other projects with proper docs and comprehensive examples.
tl;dr: The docs are beyond anemic, please do something about it.

**Comment by radex** [comment]
I understand your frustrations, but this is an open source project. I don't have a spare week to spend on crafting higher quality documentation. The time I have to dedicate to WatermelonDB I'd rather spend making it more powerful or more performant -- its core competency. If the open source community finds the documentation lacking, then the right thing to do is to contribute to the documentation. Found something really confusing and spent 2 hours trying to understand how something works? Send a quick PR with docs update so that no one else has to spend 2 hours figuring this out... Contributions are welcome ;)

**Comment by Dorza** [comment]
I'm sorry but that's a total cop out. Good documentation is essential in any serious project. If you can say you'd rather make it more power ful then others can easily say they'd rather spend time making stuff with it. Make it as powerful as you want but people still have to be able to use it and get up to speed quickly, particularly those new to use watermelondb. 

People in places of work literally don't have time or be allowed to produce documentation for your project. Other comments are correct, the documentation is lacking, incomplete and patchy leading to confusion. There also needs to be API documentation which lists all the functions and methods available. 

Flip it around...spend time doing the boring documentation and let the community make it more powerful
